### PR TITLE
Smallest distinct window problem using c++

### PR DESCRIPTION
**Pull request by vishvarana**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/smv1999/CompetitiveProgrammingQuestionBank/blob/master/CONTRIBUTING.md)?

Yes

### Description

Smallest distinct window in a string using c++ 

### Checklist

-  I've read the contribution guidelines.
- I've checked the issue list before deciding what to submit.

## Related Issues or Pull Requests

Issue #317 

## Fill up the DCP form 

If you are a part of DCP 21, fill up this [form](https://docs.google.com/forms/d/e/1FAIpQLSeRAJnHE4eAa_W0IUeNZgheiy5POX3I3tQB7DtFsiB13D5w2g/viewform) after the PR is merged.

### The links provided in the form will be checked thoroughly, if someone tries fooling around, strict actions will be taken against the individual.

### CODE REVIEW

**Review by smv1999 — CHANGES_REQUESTED** [review]
Delete all those files that are not related to this PR.

**Comment by vishvarana** [comment]
Changes made

**Review by smv1999 — CHANGES_REQUESTED** [review]
Great. Just add some comments to explain your core logic.

**Comment by vishvarana** [comment]
Added comments in cpp file

File: Strings/smallestdistinctwindow.cpp [hunk 1]
```diff
@@ -20,9 +20,9 @@ string smallestWindow(string s,string p){
 
     while(r<s.length()){
         while(r<s.length() && count<distinct){
-            freqw[s[r]]++; #taking in the main string
+            freqw[s[r]]++; //taking in the main string
             if(freqp.find(s[r])!=freqp.end() && freqw[s[r]]==freqp[s[r]]){
-                count++; #if the char matches than count increased
+                count++; //if the char matches than count increased
             }
             r++; #increment uptil the main string length
```
Comment (noviicee) [inline Strings/smallestdistinctwindow.cpp:27]: I guess there should be `//` instead of  `#` here.

Comment (noviicee) [reply Strings/smallestdistinctwindow.cpp:27]: Line 27

**Comment by vishvarana** [comment]
Changes made

**Comment by noviicee** [comment]
LGTM 🚀

**Review by smv1999 — APPROVED** [review]

**Pull request merged**